plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskList.php

```diff
@@ -350,6 +350,13 @@ public static function possibly_remove_reminder_bar() {
 	 */
 	public function get_json() {
 		$this->possibly_track_completion();
+		// Track completion of non viewable tasks.
+		foreach ( $this->tasks as $task ) {
```

Comment:
I think that would be fine, my only caveat to that would be if the `get_json` triggers other expensive logic within tasks that are not viewable, meaning it would be kinda unnecessary.
But maybe that is something the actual task should handle anyway.

Reply:
Totally.  I share the same concerns, though my thinking is that `can_view` and `is_complete` are likely the expensive queries which we now need to run in both scenarios.

Using `get_json` was kind of a workaround to trigger tracks anyway and long-term we should probably refactor this slightly to possibly track completion in the constructor now that we're doing this for all tasks.